pandas/core/generic.py

```diff
@@ -7985,33 +7985,22 @@ def _clip_with_scalar(self, lower, upper, inplace: bool_t = False):
         ):
             raise ValueError("Cannot use an NA value as a clip threshold")
 
-        mgr = self._mgr
+        result = self
+        mask = self.isna()
 
-        if inplace:
-            # cond (for putmask) identifies values to be updated.
-            # exclude boundary as values at the boundary should be no-ops.
-            if upper is not None:
-                cond = self > upper
-                mgr = mgr.putmask(mask=cond, new=upper, align=False)
-            if lower is not None:
-                cond = self < lower
-                mgr = mgr.putmask(mask=cond, new=lower, align=False)
-        else:
-            # cond (for where) identifies values to be left as-is.
-            # include boundary as values at the boundary should be no-ops.
-            mask = isna(self)
-            if upper is not None:
-                cond = mask | (self <= upper)
-                mgr = mgr.where(other=upper, cond=cond, align=False)
-            if lower is not None:
-                cond = mask | (self >= lower)
-                mgr = mgr.where(other=lower, cond=cond, align=False)
-
-        result = self._constructor(mgr)
-        if inplace:
-            return self._update_inplace(result)
-        else:
-            return result.__finalize__(self)
+        if lower is not None:
```

Comment:
This basically reverts the initial clip pr?

Reply:
> This basically reverts the initial clip pr?

Partially, but it still retains the new inplace behavior and the perf improvements for EAs.

Reply:
Thx, forgot about the inplace fix